toxcore/TCP_connection.c

```diff
@@ -1438,42 +1438,50 @@ static void do_tcp_conns(const Logger *logger, TCP_Connections *tcp_c, void *use
 
 static void kill_nonused_tcp(TCP_Connections *tcp_c)
 {
-    if (tcp_c->tcp_connections_length == 0) {
+    const uint32_t num_tcp_connections = tcp_c->tcp_connections_length;
```

Comment:
nit: I'd un-extract this variable. Extracting it makes it slightly less obvious that this number comes from tcp_c:

```c
    for (uint32_t i = 0; i < num_tcp_connections; ++i) {
        TCP_con *tcp_con = get_tcp_connection(tcp_c, i);
```

vs.

```c
    for (uint32_t i = 0; i < tcp_c->tcp_connections_length; ++i) {
        TCP_con *tcp_con = get_tcp_connection(tcp_c, i);
```

Having the variable doesn't save a whole lot, but makes it less locally-correct. What we're expressing is "for each tcp connection". Using a variable from somewhere else that happens to represent the count breaks apart the "foreach" loop.